### PR TITLE
🌱 GitLab: Add log message

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Adds log message when GitLab is running on non-HEAD that scores may be inaccurate.

#### Which issue(s) this PR fixes
Temporary warning #3193 to unblock release.

### CODE REVIEW

File: clients/repo_client.go [hunk 1]
```diff
@@ -38,6 +45,8 @@ type RepoClient interface {
 	LocalPath() (string, error)
 	GetFileContent(filename string) ([]byte, error)
 	GetBranch(branch string) (*BranchRef, error)
+	GetCommitSHA() string
```
Comment (spencerschrock) [inline clients/repo_client.go:48]: I don't think this is a property of the repo client, but rather the repo right?

File: clients/repo_client.go [hunk 2]
```diff
@@ -38,6 +45,8 @@ type RepoClient interface {
...
+	GetPlatform() string
```
Comment (spencerschrock) [inline clients/repo_client.go:49]: this is a breaking change, do we know if allstar is still the only implementer of custom clients? What about legitify?

**Review by spencerschrock — COMMENTED** [review]
Is this something we want in the results? Or just when running the CLI?

File: clients/repo_client.go (hunk 2 above)
Comment (raghavkaul) [reply clients/repo_client.go:49]: Oh right... what do you think of just a bare `fmt.Printf` in `clients/gitlabrepo/commits.go`? I had that originally but it felt janky so I did this whole thing

File: clients/repo_client.go (hunk 1 above)
Comment (raghavkaul) [reply clients/repo_client.go:48]: Yeah but repos aren't accessible in `checks`

File: clients/repo_client.go (hunk 2 above)
Comment (spencerschrock) [reply clients/repo_client.go:49]: It's a little hacky, yeah. At the minimum, we need to ensure it's printed to stderr. Printing to stdout breaks scorecard action at the very least

**Review by spencerschrock — COMMENTED** [review]
Can we add a TODO comment with tracking issue?

Note: this can run multiple times, but not sure pushing the error deeper can be done as easily as right here

```console
SCORECARD_EXPERIMENTAL=1 go run main.go --repo gitlab.com/ossf-test/scorecard-archived-project --commit 240e1fbc4c4ea614527c59f453bf54107d8833f2
...
Scorecard may be missing merge requests when running on non-HEAD of a GitLab repo. Code-Review scores may be lower.
Scorecard may be missing merge requests when running on non-HEAD of a GitLab repo. Code-Review scores may be lower.
Scorecard may be missing merge requests when running on non-HEAD of a GitLab repo. Code-Review scores may be lower.
Scorecard may be missing merge requests when running on non-HEAD of a GitLab repo. Code-Review scores may be lower.
...
```

**Review by spencerschrock — APPROVED** [review]

**Pull request merged**